### PR TITLE
Speed up `html-proofer`  test 

### PR DESCRIPTION
**Pull request by ganezasan**
# Description

- use `parallel` feature in `html-proofer` to speed up test
- use medium+ resouce_class in CircleCI
- export `html-proofer test` and `jekyll build` commands into Rakefile
- pack artifacts before uploading them into CircleCI artifacts

# Reasons

The `html-proofer` test generally takes 4 to 5 mins, so I customized it to use `parallel` feature in  `html-proofer` to speed up this test. Also, I changed circleci.yml to use `medium+` resource_class, and now the `html-proofer` test tasks about 1 min.

> https://app.circleci.com/pipelines/github/circleci/circleci-docs/27419/workflows/0309440a-a3fe-4bc0-8e5e-9667ab5002e6/jobs/75692/parallel-runs/0/steps/0-120

Also, the html-proofer and jekyll build commands were complexed. And it prevents running these commands locally.
So, I exported `html-proofer test` and `jekyll build` commands into Rakefile. 

```
$ bundle exec rake build
$ bundle exec rake test
```

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+task :make_mock_api do
+  puts 'Make mock API docs'
+  sh "echo \"<html lang=''><head><link href='/docs/assets/img/icons/favicon.png' rel='icon' type='image/png' /></head></html>\" > ./jekyll/_site/index.html"
+  sh "mkdir -p ./jekyll/_site/docs/api/2.0/"
+  sh "mkdir -p ./jekyll/_site/docs/api/v2/"
+  sh "cp ./jekyll/_site/index.html ./jekyll/_site/docs/api/2.0/index.html"
+  sh "cp ./jekyll/_site/index.html ./jekyll/_site/docs/api/v2/index.html"
+  sh "cp ./jekyll/_site/index.html ./jekyll/_site/docs/api/index.html"
+  sh "rm ./jekyll/_site/index.html"
+end
```
Comment (teesloane) [inline Rakefile:36]: What is :make_mock_api used for / where is it being implemented in the `config.yml`?

**Comment by ganezasan** [comment]
@teesloane I'm glad to hear that!

> 1. Can you confirm that this won't change our deploy system (I don't know if we use artifacts to deploy, but I noticed there was a change to it). If you need to test on a staging site, you can re-create this branch so that it ends in `-preview` to view a live deployment (ex: `fix-speedup-ci-test-preview`.

I'll double-check this to create another branch.

> 2. I haven't used Rake before - would it be idiomatic to migrate some of our smaller build scripts in the `scripts` folder into rake procedures rather than using the rake file to call the files? Does it make sense to aggregate things into one place?

Rakefile is the ruby version of the Makefile. We can call the script files from Rakefile, and I don't think it's not necessary to migrate all the build scripts into Rakefile, but it worth migrating smaller build scripts to Rakefile. Also, I think it'll be a good idea to define all the tasks in Rakefile.

```
bundle exec rake -T 
rake build              # Build Jekyll site
rake build_local        # Build your jekyll site for local env
rake make_mock_api      # Make mock API docs
rake shim_untranslated  # Shim untranslated Japanese pages
rake test               # Test with HTMLproofer
```

At this time, `html-proofer` doesn't export the `parallel` option, so we need to manually import that function, and set the option.
> https://github.com/gjtorikian/html-proofer/blob/main/bin/htmlproofer

File: Rakefile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+task :test do
+  puts 'Test with HTMLproofer'
+  ignore_files = ["./jekyll/_site/#{JEKYLL_BASENAME}/api/v2/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/runner-installation/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/security-server/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/v.2.19-overview/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/customizations/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/aws-prereq/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/ops/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/about-circleci/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/demo-apps/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/google-auth/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/orb-concepts/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/ja/2.0/tutorials/index.html","./jekyll/_site/#{JEKYLL_BASENAME}/reference-2-1/index.html"]
```
[the remark this answers is not in the document]
Comment (ganezasan) [reply Rakefile:45]: Right! I'll clean up this!

File: Rakefile (hunk 1 above)
Comment (ganezasan) [reply Rakefile]: To successfully run the `html-proofer` test locally, we need to build API as well, but I want to skip that step.
So, this is a workaround to skip building API docs. 

I created another task `build_local`, this is the command for preparing `html-proofer` test.
https://github.com/circleci/circleci-docs/pull/5141/files#diff-ee98e028c59b193d58fde56ab4daf54d43c486ae674e63d50ddf300b07943e0fR38

**Comment by ganezasan** [comment]
@teesloane I've checked the staging site of this change, and it seems there is no problem.

> https://app.circleci.com/pipelines/github/circleci/circleci-docs/27427/workflows/002253ab-2bef-406d-9793-4b249a5e8669/jobs/75723
> http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/fix-speedup-ci-test-preview/

**Review by teesloane — APPROVED** [review]
Thanks again @ganezasan !

**Pull request merged**